### ISSUE TITLE
Try bfio

### ISSUE THREAD

**Issue by joshmoore** [ome/ome_zarr_test_suite#14]
see: https://pypi.org/project/bfio/

**Comment by Nicholas-Schaub** [comment]
Hey, I am the lead developer of `bfio`. I would be happy to contribute in any way. We are in the process of upgrading our ome zarr spec. It's not a complete implementation, but we are updating the way we handle metadata since there wasn't a clear approach to doing it when we first created it.

**Comment by joshmoore** [comment]
Hey @Nicholas-Schaub! Thanks for stopping by. A first question: do you have reader & writer implementations?

**Comment by Nicholas-Schaub** [comment]
Yes, but they are from v0.1. We are in the process of updating now. When that spec came out, the Bioformats raw converter was just dumping the ome xml into the same directory, which did not work for us. We ended up storing the metadata as zarr attributes, but I see that you are now storing the file inside zarr, which I think is an excellent solution.

Also, our implementation is not a complete implementation. We don't support things like the hcs structure or multiple images. That's primarily because we don't have a need for them on our platform. I think based on the files you have here, I think our implementation (when finished updating) should pass.

**Comment by Nicholas-Schaub** [comment]
@joshmoore Hey. I'm starting to integrate this as a submodule into bfio, develop unit tests, etc. Should `v0.3` and `v0.4` data have OME XML? It appears as though `v0.1` and `v0.2` both have OME XML, but these are missing for the other versions.

Also, I don't see any indication of data dimensionality/layout in any of the metadata for `v0.3` and `v0.4`.

Is OME XML missing from the OME NGFF spec? Is this something that is going to be added? Is there some convention you plan on using even if not formally defined in the spec?

**Comment by joshmoore** [comment]
> Should `v0.3` and `v0.4` data have OME XML?

Hi @Nicholas-Schaub. `bioformats2raw` has produced the OME-XML for most if not all versions but it's not considered part of the specification to date. https://github.com/ome/ngff/issues/104 is looking to add that support. I'd assume that an implementation SHOULD (per [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119)) ignore the OME-XML. The result of that is that an implementation then treats individually numbered subdirectory as a standalone OME-Zarr.

> Is OME XML missing from the OME NGFF spec? Is this something that is going to be added? Is there some convention you plan on using even if not formally defined in the spec?

Yes. Yes. Yes. 🙂 per 104.
